app.rb

```diff
@@ -6,16 +6,54 @@
 $polls = [
   Poll.new('好きな料理', ['肉じゃが', 'しょうが焼き', 'から揚げ']),
   Poll.new('人気投票', ['おむすびけん', 'クックパッドたん']),
+  Poll.new('期限を過ぎた投票', ['a', 'b'], Time.now)
 ]
 
 get '/' do
   erb :index, locals: { polls: $polls }
 end
 
+get '/polls/new' do
+  erb :new_poll
+end
+
+post '/polls/new' do
+  halt 404, '選択肢を2つ以上入力してください' if params['candidates'].length < 1
```

Comment:
エラーメッセージと条件判定が合ってなさそう？
また、実際にフォームを空のまま入力すると `params['candidates']` は空文字列が3要素入ってしまうので、空文字列を弾く処理も必要そうでした。

```suggestion
  candidates = Array(params['candidates']).reject { |candidate| candidate.nil? || candidate.empty?  }
  halt 404, '選択肢を2つ以上入力してください' if params['candidates'].length < 2
```